cpp/src/ray/runtime/abstract_ray_runtime.cc

```diff
@@ -53,13 +53,14 @@ std::shared_ptr<AbstractRayRuntime> AbstractRayRuntime::DoInit() {
     ProcessHelper::GetInstance().RayStart(TaskExecutor::ExecuteTask);
     runtime = std::shared_ptr<AbstractRayRuntime>(new NativeRayRuntime());
     RAY_LOG(INFO) << "Native ray runtime started.";
-    if (ConfigInternal::Instance().worker_type == WorkerType::WORKER) {
-      // Load functions from code search path.
-      FunctionHelper::GetInstance().LoadFunctionsFromPaths(
-          ConfigInternal::Instance().code_search_path);
-    }
   }
   RAY_CHECK(runtime);
+  internal::RayRuntimeHolder::Instance().Init(runtime);
+  if (ConfigInternal::Instance().worker_type == WorkerType::WORKER) {
```

Comment:
Can we move this block to native runtime? Do we need this for single process?

Reply:
I move "internal::RayRuntimeHolder::Instance().Init(runtime)" from  api.cc to abstract_ray_runtime.cc, so single mode  api_test also need to init the runtime.
Why move into abstract_ray_runtime.cc? because i need it before LoadFunctionsFromPaths(init ray runtime in so)